### PR TITLE
Feature add status to service and service tree structs 

### PR DESCRIPTION
**Pull request by Poltergeist**
This adds getStatus to the service and the serviceTree structs as this is mandatory for the status bars on the service table. It does only support a small set of statuses to keep this PR small and the rest will be issued in another PR later.

### CODE REVIEW

**Comment by colin-msphere** [comment]
retest this please

File: src/js/structs/ServiceTree.js [hunk 1]
```diff
@@ -91,6 +102,16 @@ module.exports = class ServiceTree extends Tree {
     });
   }
 
+  getInstances() {
```
Comment (kennyt) [inline src/js/structs/ServiceTree.js]: can we name this `getInstanceCount` or `getInstancesCount`? (I don't know how to english) since this is a number

File: src/js/structs/__tests__/ServiceTree-test.js [hunk 2]
```diff
@@ -224,6 +225,67 @@ describe('ServiceTree', function () {
...
+
+      expect(serviceTree.getDeployments()).toEqual([]);
+    });
+
+    it('should return an array with three deployments', function () {
+      let serviceTree = new ServiceTree(
+        {
+          id: '/group/id',
+          apps: [
+            {id: '/alpha', cmd: 'cmd', deployments: [{id: "4d08fc0d-d450-4a3e-9c85-464ffd7565f2"}]},
```
Comment (mlunoe) [inline src/js/structs/__tests__/ServiceTree-test.js]: You have a bunch of double quote strings. Let's change these to single quotes

File: src/js/structs/__tests__/ServiceTree-test.js [hunk 3]
```diff
@@ -224,6 +225,67 @@ describe('ServiceTree', function () {
...
+              cmd: 'cmd',
+              deployments: [{id: "4d08fc0d-d450-4a3e-9c85-464ffd7565f3"}],
+              labels: {DCOS_PACKAGE_FRAMEWORK_NAME: 'beta'}
+            },
+            {id: '/gamma', cmd: 'cmd', labels: {RANDOM_LABEL: 'random'}}
+          ],
+          groups: [
+            {
+              id: '/test', apps: [
+              {id: '/foo', cmd: 'cmd', deployments: [{id: "4d08fc0d-d450-4a3e-9c85-464ffd7565f1"}]},
```
Comment (mlunoe) [inline src/js/structs/__tests__/ServiceTree-test.js]: Indentation for the next 3 lines are wrong

File: src/js/structs/__tests__/ServiceTree-test.js [hunk 4]
```diff
@@ -224,6 +225,67 @@ describe('ServiceTree', function () {
 
   });
 
+  describe('#getDeployments', function () {
```
Comment (mlunoe) [inline src/js/structs/__tests__/ServiceTree-test.js]: This was not you but there is an extra newline on line 213

File: src/js/structs/__tests__/ServiceTree-test.js (hunk 3 above)
Comment (mlunoe) [inline src/js/structs/__tests__/ServiceTree-test.js]: This is duplicated throughout this file

File: src/js/structs/Service.js [hunk 5]
```diff
@@ -81,12 +86,32 @@ module.exports = class Service extends Item {
...
+  }
+
   getTasksSummary() {
     return {
       tasksHealthy: this.get('tasksHealthy'),
       tasksRunning: this.get('tasksRunning'),
       tasksStaged: this.get('tasksStaged'),
-      tasksUnhealthy: this.get('tasksUnhealthy')
+      tasksUnhealthy: this.get('tasksUnhealthy'),
+      tasksUnknown: this.get('tasksRunning') -
```
Comment (mlunoe) [inline src/js/structs/Service.js]: What do we need this for? Isn't this just an agregate of the others? Seems weird to introduce

File: src/js/structs/__tests__/ServiceTree-test.js [hunk 6]
```diff
@@ -397,7 +545,8 @@ describe('ServiceTree', function () {
         tasksStaged: 1,
         tasksRunning: 4,
         tasksHealthy: 16,
-        tasksUnhealthy: 1
+        tasksUnhealthy: 1,
+        tasksUnknown: -13
```
Comment (mlunoe) [inline src/js/structs/__tests__/ServiceTree-test.js]: Also, it seems weird that this returns a negative number

**Comment by mlunoe** [comment]
A couple of comments, but I think in general it looks good!

File: src/js/structs/Service.js (hunk 5 above)
Comment (Poltergeist) [inline src/js/structs/Service.js]: Yes it is a number which is only aggregated by the other statuses but we will need it at multiple places where the status counts are used. I thought the best option was to implement it here instead of duplicating the code all the time. This state is used for applications which a) have no health checks defined, or b) did not become healthy yet.

File: src/js/structs/__tests__/ServiceTree-test.js (hunk 6 above)
Comment (Poltergeist) [inline src/js/structs/__tests__/ServiceTree-test.js]: It is weird, but this example will actually never happen, tasksHealthy is always smaller or equal to tasksRunning. I'll change the example to be more representing the real world.

File: src/js/structs/ServiceTree.js (hunk 1 above)
Comment (orlandohohmeier) [inline src/js/structs/ServiceTree.js]: Let's do this refactoring in a different PR. The `getInstances` method only implements the current service struct interface and we should refactor both at a stroke.

File: src/js/structs/Service.js (hunk 5 above)
Comment (orlandohohmeier) [inline src/js/structs/Service.js]: I don't see any reason not to add aggregated data, as long as it is consistent and *useful*. In my humble opinion are structs (at least in our case) meant to typecast data, provide an abstraction layer, and a nice and easy to use interface.

**Comment by orlandohohmeier** [comment]
L:eyes:ks good to me!

File: src/js/structs/Service.js (hunk 5 above)
Comment (mlunoe) [inline src/js/structs/Service.js]: That's fair point :)

**Comment by mlunoe** [comment]
Reviewer: @mlunoe
